specs/unified_kernel_image.md

```diff
@@ -159,3 +159,36 @@ that tools staging or consuming UKIs have a common place to store and look for t
 
 The installed UKIs should have a filename `<version format specification>.efi`, i.e. the filename is left to
 implementers but must be valid for comparisons according to the [Version Format Specification](version_format_specification.md).
+
+### Auxiliary resources
+
+Auxiliary UKI resources (such as PE addons for command line extensions and similar,
+as well as systemd-sysext and systemd-confext DDIs) built centrally by distributions and
+installed via package manager should be installed into locations depending on
+whether they should be applied to all UKIs installed in the ESP, or only to a
+single specific UKI.
+
+UKI auxiliary resources that shall apply to *all* installed UKIs may be
+installed into `/usr/lib/modules/uki.extra.d/`. UKI auxiliary resources that
+shall apply to *one* specific installed kernel may be instead installed into
```

Comment:
I'm not entirely sure about the requirement level, but *may* is weaker that *should*. Should this instead be
```suggestion
UKI auxiliary resources that shall apply to *all* installed UKIs should be
installed into `/usr/lib/modules/uki.extra.d/`. UKI auxiliary resources that
shall apply to *one* specific installed kernel should be instead installed into
```
since *may* is an optional thing, but this should probably be stronger?

To avoid confusion, I'd probably also just drop the "shall" in front of "apply" here.